### PR TITLE
Visualization of a SAR image as a ground surface added

### PR DESCRIPTION
**Pull request by MargrietGroenendijk**
First iteration of this type of image, which uses a blob to be able to display the data.  This PR will make it possible to test this on TileDB Cloud in the next deployment. 

Example API call to create this new visualization:

```
show.from_array(array_uri="autzen-sar-gec",
                style = 'ground',
                band = band,
                xy_bbox = [11500,14000,11500,14000],
                attribute = 'TDB_VALUES',
                scale_factor = scale_factor,
                width = 1000,
                height = 800,
                inspector = True)
```

### CODE REVIEW

File: pybabylonjs/babylonjs.py [hunk 1]
```diff
@@ -68,6 +68,8 @@ def set_defaults(validator, properties, instance, schema):
 
 mbrs_schema = deepcopy(core_schema)
 
+ground_schema = {"type": "object"}
```
[the remark this answers is not in the document]
Comment (MargrietGroenendijk) [reply pybabylonjs/babylonjs.py]: done, this is now as the below. Also added `image_type` to be able to add different types of data later. 

```python
ground_schema = {
    "type": "object",
    "properties": {
        "inspector": {"type": "boolean", "default": False},
        "image_type": {"type": "string", "default": "general"},
        "width": {"type": "number", "default": 800},
        "height": {"type": "number", "default": 600},
        "z_scale": {"type": "number", "default": 1},
        "wheel_precision": {"type": "number", "default": -1},
        "xy_bbox": {"type": "array", "default": [0, 1, 0, 1]},
        "band": {"type": "number", "default": 1},
        "scale_factor": {"type": "number", "default": 1},
        "img_width": {"type": "number"},
        "img_height": {"type": "number"},
    },
    "required": ["xy_bbox", "img_width"],
}
```

File: pybabylonjs/data.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+    img_nr: image number to use as ground image
+    scale_factor: factor to scale the values in the image
+    """
+
+    def numpy_to_binary(arr):
+        is_success, buffer = cv2.imencode(".png", arr)
+        io_buf = io.BytesIO(buffer)
+        return io_buf.read()
+
+    bbox = kwargs["bbox"]
```
[the remark this answers is not in the document]
Comment (MargrietGroenendijk) [reply pybabylonjs/data.py]: I added this, see the schema above

File: pybabylonjs/data.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    nr = kwargs["img_nr"]
+
+    with tiledb.open(array_uri, "r") as arr:
+        img = arr[nr, bbox[0] : bbox[1], bbox[2] : bbox[3]][kwargs["attribute"]]
+
+    img_norm = 20 * np.log10(img * kwargs["scale_factor"])
```
[the remark this answers is not in the document]
Comment (MargrietGroenendijk) [reply pybabylonjs/data.py]: Solved this by adding a parameter `image_type` that only does this conversion when `image_type == "sar"` 

Renamed `scale_factor` to `sar_scale_factor` to make this clearer as well

File: src/widget.ts [hunk 4]
```diff
@@ -260,16 +261,16 @@ export class BabylonMBRSView extends BabylonBaseView {
       const maxz = extents[5];
 
       const scale = this.zScale;
-
+
```
[the remark this answers is not in the document]
Comment (MargrietGroenendijk) [reply src/widget.ts:264]: removed

**Review by normanb — APPROVED** [review]

**Pull request merged**